app/grandchallenge/evaluation/admin.py

```diff
@@ -58,6 +65,84 @@ def clean(self):
         return cleaned_data
 
 
+class ConfigureAlgorithmPhasesForm(Form):
+    phases = ModelMultipleChoiceField(
+        queryset=Phase.objects.select_related("challenge")
+        .filter(submission_kind=SubmissionKindChoices.CSV)
+        .all()
+    )
+    algorithm_inputs = ModelMultipleChoiceField(
+        queryset=ComponentInterface.objects.all()
+    )
+    algorithm_outputs = ModelMultipleChoiceField(
+        queryset=ComponentInterface.objects.all()
+    )
```

Comment:
Could we make these the autocomplete fields like we have on the normal form, these are too fiddly with so many interfaces. You should be able to reuse how the admin does it, I think:

<img width="540" alt="Screenshot 2024-02-29 at 15 46 32" src="https://github.com/comic/grand-challenge.org/assets/12661555/9d206368-b3a0-491f-9751-084a1630d44d">

Reply:
I can get the autocomplete widget to work for the inputs and outputs, but not for the phases.

Reply:
Okay, then I think I prefer `CheckboxSelectMultiple` over `SelectMultiple` as it's nicer to use (feel like too many mistakes can creep in with cmd+click+scroll), but up to you.